decko/rails/controllers/card_controller.rb

```diff
@@ -49,37 +49,43 @@ def asset
 
   before_action :setup, except: [:asset]
   before_action :authenticate, except: [:asset]
-  before_action :load_id, only: [:read]
+  before_action :load_mark, only: [:read]
   before_action :load_card, except: [:asset]
+  before_action :load_action, only: [:read]
   before_action :refresh_card, only: [:create, :update, :delete]
 
   def setup
     Card::Machine.refresh_script_and_style unless params[:explicit_file]
     Card::Cache.renew
     Card::Env.reset controller: self
-    request.format = :html if Card::Env.ajax? && !params[:format]
   end
 
   def authenticate
     Card::Auth.set_current params[:token], params[:current]
   end
 
-  def load_id
-    params[:id] = interpret_id params[:id]
+  def load_mark
+    params[:mark] = interpret_mark params[:mark]
   end
 
   def load_card
-    @card = Card.controller_fetch params
-    raise Card::Error::NotFound unless @card
-    load_action
-    record_as_main
-    card.errors.any? ? render_errors : true
+    handle_errors do
+      @card = Card.controller_fetch params
+      raise Card::Error::NotFound unless card
+      record_as_main
+    end
   end
 
   def load_action
-    @card.select_action_by_params params
+    handle_errors do
```

Comment:
I changed this logic a bit.  we don't need to call load_action when we're loading cards for create/update/delete, do we?